include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -345,29 +345,57 @@ typedef enum OrtCudnnConvAlgoSearch {
 */
 typedef struct OrtCUDAProviderOptions {
 #ifdef __cplusplus
-  OrtCUDAProviderOptions() : device_id{}, cudnn_conv_algo_search{OrtCudnnConvAlgoSearchExhaustive}, gpu_mem_limit{SIZE_MAX}, arena_extend_strategy{}, do_copy_in_default_stream{}, has_user_compute_stream{}, user_compute_stream{}, default_memory_arena_cfg{} {}
+  OrtCUDAProviderOptions() : device_id{}, cudnn_conv_algo_search{OrtCudnnConvAlgoSearchExhaustive}, gpu_mem_limit{SIZE_MAX}, arena_extend_strategy{}, do_copy_in_default_stream{1}, has_user_compute_stream{}, user_compute_stream{}, default_memory_arena_cfg{} {}
 #endif
 
-  int device_id;  ///< CUDA device id (0 = default device)
+  /** \brief CUDA device Id
+  *   Defaults to 0.
```

Comment:
identifying the default values is good. does it only apply to C++ usage though?

Reply:
No, it is not limited to just C++.

Reply:
was mainly wondering about the C API, as the constructor is within `#ifdef __cplusplus`

Reply:
Can we have constructors for structs in C ? My understanding is for the users of the C API, the users are expected to explicitly initialize the struct's members.

Reply:
that's my understanding too. would it be useful to clarify that the default value doesn't apply to C usage, as this is in the C API header?
or should we provide a way to get a structure with these default values that's also compatible with C?

Reply:
I agree that there should be a way to fetch a default initialized instance for the convenience of the C API users. But can we maintain status quo for now and add the enhancement later ? Clarifying with a comment for this release is a nice middle ground. What do you think ?

Reply:
sure, we can add it later